lib/matplotlib/tests/test_datetime.py

```diff
@@ -450,11 +450,27 @@ def test_pcolorfast(self):
         fig, ax = plt.subplots()
         ax.pcolorfast(...)
 
-    @pytest.mark.xfail(reason="Test for pcolormesh not written yet")
     @mpl.style.context("default")
     def test_pcolormesh(self):
-        fig, ax = plt.subplots()
-        ax.pcolormesh(...)
+        fig, (ax1, ax2, ax3, ax4) = plt.subplots(4, 1, layout="constrained")
+
+        np.random.seed(19680801)
+
+        x_base_date = datetime.datetime(2023, 1, 1)
+        x_dates = [x_base_date + datetime.timedelta(days=i) for i in range(5)]
+
+        y_base_date = datetime.datetime(2023, 1, 1)
+        y_dates = [y_base_date + datetime.timedelta(days=i) for i in range(5)]
+
+        x_timestamps = [date.timestamp() for date in x_dates]
+        y_timestamps = [date.timestamp() for date in y_dates]
```

Comment:
I would avoid using the timestamp for the non-date data, as it is a number that looks like it should be interpreted as a data (and thus made me do a doubletake when I saw the image)

Perhaps just do e.g. `np.linspace(0, 10, x_dates.size)` or something similar